templates/template.rb

```diff
@@ -417,6 +419,31 @@ def initialize(config, flags)
             # changed to use fetch instead of delete.
             comment = options.delete(:comment)
 
+            if kinds = options[:kind]
+              kinds = [kinds] unless kinds.is_a?(Array)
+              kinds = kinds.map do |kind|
+                case kind
+                when "non-void expression"
+                  # the actual list of types would be way too long
+                  "Node"
+                when "pattern expression"
+                  # TODO: can we be more precise, is the list small enough?
+                  "Node"
```

Comment:
FWIW this is the full list I got from running the test suite:
```ruby
AlternationPatternNode
ArrayPatternNode
CapturePatternNode
FindPatternNode
HashPatternNode
PinnedExpressionNode
PinnedVariableNode
LocalVariableTargetNode

ImplicitRestNode
SplatNode

IfNode
UnlessNode

SymbolNode
InterpolatedSymbolNode
TrueNode
FalseNode
NilNode
IntegerNode
FloatNode
RangeNode
StringNode
InterpolatedStringNode
XStringNode
RegularExpressionNode
InterpolatedRegularExpressionNode
SourceFileNode
SourceLineNode
SourceEncodingNode
ParenthesesNode
ArrayNode
RationalNode
ImaginaryNode

SelfNode
ConstantReadNode
ConstantPathNode
LambdaNode

MissingNode
```
A 37-wide union type is probably not very practical so I'll leave it as `Node` here and remove the TODO